src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -6582,13 +6587,14 @@ instruct cmovP_regUCF2_ne(cmpOpUCF2 cop, rFlagsRegUCF cr, rRegP dst, rRegP src)
 instruct cmovP_regUCF2_ne_ndd(cmpOpUCF2 cop, rFlagsRegUCF cr, rRegP dst, rRegP src1, rRegP src2) %{
   predicate(UseAPX && n->in(1)->in(1)->as_Bool()->_test._test == BoolTest::ne);
   match(Set dst (CMoveP (Binary cop cr) (Binary src1 src2)));
+  effect(TEMP dst);
 
   ins_cost(200);
   format %{ "ecmovpq  $dst, $src1, $src2\n\t"
-            "ecmovneq $dst, $src1, $src2" %}
+            "cmovneq  $dst, $src2" %}
   ins_encode %{
     __ ecmovq(Assembler::parity, $dst$$Register, $src1$$Register, $src2$$Register);
-    __ ecmovq(Assembler::notEqual, $dst$$Register, $src1$$Register, $src2$$Register);
+    __ cmovq(Assembler::notEqual, $dst$$Register, $src2$$Register);
```

Comment:
FTR, 
Instruction sequence for NaN-APX:-
 
CMOVE DST , SRC if FLAG is PF
CMOVE DST , SRC if FLAG is NE
 
With APX

CMOVE DST, DEF_DST, SRC if FLAG is PF
CMOVE DST, DEF_DST, SRC if FLAT is NE

Root cause:
So in back-to-back CMOVEs we end up updating the DST with DEF_DST if ZF is set since NE is false and not retain the DST value updated by prior CMOVE, this is not the case with Non-APX case since second DST is not executed is ZF is set and DST retains the value set by prior CMOVE.